dataland-e2etests/src/test/kotlin/org/dataland/e2etests/tests/CompanyDataControllerGetCompaniesEndpointTest.kt

```diff
@@ -0,0 +1,287 @@
+        // data having non-unique identifier values for different identifier types)
+        searchResponse.retainAll {
+            it.companyInformation.identifiers.any {
+                    identifierInResponse ->
+                identifierInResponse.identifierType == identifier.identifierType
+            }
+        }
+        assertTrue(
+            searchResponse.all { results -> results.companyInformation.identifiers.any { it == identifier } },
+            "The search by identifier returns at least one company that does not contain the looked" +
+                "for value $identifier.",
+        )
+    }
+
+    @Test
+    fun `search for all identifier values and check if all results contain the looked for value`() {
+        apiAccessor.jwtHelper.authenticateApiCallsWithJwtForTechnicalUser(TechnicalUser.Uploader)
+        val testString = UUID.randomUUID().toString()
+        val testCompanyList = listOf(
+            CompanyInformation(
+                company1, "",
+                listOf(
+                    apiAccessor.createCompanyIdentifier(CompanyIdentifier.IdentifierType.isin, "Isin$testString"),
+                    apiAccessor.createCompanyIdentifier(CompanyIdentifier.IdentifierType.lei, "Lei$testString"),
+                ),
+                "",
+                listOf(company1 + testString),
+            ),
+        )
+        for (company in testCompanyList) {
+            val companyResponse = apiAccessor.companyDataControllerApi.postCompany(company)
+            uploadTestDataSet(companyResponse.companyId)
+            for (identifier in company.identifiers) {
+                testThatSearchForCompanyIdentifierWorks(identifier)
+            }
+        }
+    }
+
+    @Test
+    fun `retrieve companies as a list and check for each company if it can be found as expected`() {
+        apiAccessor.jwtHelper.authenticateApiCallsWithJwtForTechnicalUser(TechnicalUser.Uploader)
+        val companyIdentifier = listOf(
+            apiAccessor.createCompanyIdentifier(CompanyIdentifier.IdentifierType.lei, UUID.randomUUID().toString()),
+        )
+        val companyInformation = CompanyInformation(
+            "retrieve empty search string", "", companyIdentifier, "",
+            listOf(),
+        )
+        val uploadedCompany = apiAccessor.companyDataControllerApi.postCompany(companyInformation)
+        val uploadedData = uploadTestDataSet(uploadedCompany.companyId).copy(uploaderUserId = null)
+        val expectedCompany = StoredCompany(
+            uploadedCompany.companyId,
+            uploadedCompany.companyInformation,
+            listOf(uploadedData),
+        )
+
+        val retrievedCompanies = apiAccessor.getCompaniesOnlyByName("")
+        assertTrue(
+            retrievedCompanies.contains(expectedCompany),
+            "Not all the companyInformation of the posted companies could be found in the stored companies.",
+        )
+    }
```

Comment:
The wording of this test implies that many companies are being uploaded, while it's in fact only one company which is uploaded.

Reply:
To be pushed